### PR TITLE
EOS 17507:  s3 URL field from S3 accounts and IAM user tabs hidden if not correctly created

### PR DESCRIPTION
**Pull request by Vrishali-web-dev**
# UI

 

 S3 UI : Show S3 URLs only if correctly created and hide if not. 

 

## Problem Statement
<pre>
  <code>
    Story Ref (if any): EOS-17507 UI: S3 use cases should be working on VM through CSM (FE)
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  No
  </code>
</pre>
## Problem Description
<pre>
  <code>

- S3 features are broken from CSM side. This might be due to cluster IP is not available for VM. Fix all the issues and make all the features of S3 working on VM.
- Insure that all S3 features and functionalities working properly.
- Remove S3 URL field if no proper URL is created(S3 accounts and I am user tab).

  </code>
</pre>
## Solution/Screenshots
 

- S3 user is able to create and delete I am user account
- S3 user is able to edit account
- S3 user is able to generate and delete access key
- S3 user is able to create, edit and delete bucket 
- S3 URL is not visible if not created properly on S3 account and I am user tabs

S3 URL filed is hidden:
image

image

Access key is generated successfully:
image

I am user account created successfully :
image

Tooltip for bucket URL is removed:
image

<pre>
  <code>
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
**S3 account:**
- S3 URL field is hidden.
- S3 accounts created successfully.
- Access key generated successfully.

**IAM User:**
- S3 URL field is hidden.
- IAM Users created successfully.
- Access key generated successfully.

**Bucket:**
- Buckets created successfully.
- Bucket URL field is hidden.

  </code>

Signed-off-by: Vrishali Danave <vrishali.danave@seagate.com>

### CODE REVIEW

File: gui/src/components/s3/account-management.vue [hunk 1]
```diff
@@ -700,7 +700,8 @@ export default class CortxAccountManagement extends Vue {
     const res: any = await Api.getAll(apiRegister.s3_account);
     this.accountsList = res && res.data ? res.data.s3_accounts : [];
     this.s3Url = res.data && res.data.s3_urls ? res.data.s3_urls : [];
-    if (this.s3Url[0] === "http://None") {
+    
+    if (this.s3Url.length && this.s3Url[0] === "http://None") {
```
Comment (kupranay) [inline gui/src/components/s3/account-management.vue:704]: please check for this.s3URL is undefined or not

Comment (kupranay) [reply gui/src/components/s3/account-management.vue:704]: naming convetion not followed. s3URL should be used as isS3URL

File: gui/src/components/s3/bucket-creation.vue [hunk 2]
```diff
@@ -377,6 +377,7 @@ export default class CortxBucketCreation extends Vue {
   private bucketNameTooltipMessage = bucketNameTooltipMessage;
   private bucketUrl = "";
   private noBucketPolicy: boolean;
+  private bucketUrlNone: boolean = false;
```
Comment (kupranay) [inline gui/src/components/s3/bucket-creation.vue:380]: variable name should suggest the type and purpose.

**Review by kupranay — CHANGES_REQUESTED** [review]

File: gui/src/components/s3/account-management.vue (hunk 1 above)
Comment (Vrishali-web-dev) [reply gui/src/components/s3/account-management.vue:704]: The check is already there at line 702.

File: gui/src/components/s3/bucket-creation.vue (hunk 2 above)
Comment (Vrishali-web-dev) [reply gui/src/components/s3/bucket-creation.vue]: Variable names are updated

**Review by shrimetta — APPROVED** [review]

**Review by shrimetta — APPROVED** [review]

**Pull request merged**